src/http/routes.cc

```diff
@@ -140,6 +143,13 @@ routes& routes::add(operation_type type, const url& url,
     return add(rule, type);
 }
 
+routes& routes::add_default_handler(handler_base* handler) {
+    if(_default_handler == nullptr) {
+        _default_handler = handler;
+    }
```

Comment:
I don't understand the if... Why not just set _default_handler without the if?
If you, for some reason, want to forbid overriding a previously-set default, please throw an exception. Don't just silently ignore the call.

Reply:
The exception will cause another thread that wants the same handler to be used to throw an exception, hence the if block, do let me know, I can remove the if clause though without any unwanted side-effects.